commit 32629c175fe8392de9517604b90dd72365f84968
Author: Jan Alexander Steffens (heftig) <jan.steffens@gmail.com>
Date:   Sun Jun 2 09:59:57 2013 -0700

    Type problems prevent compilation with GCC 4.8 on i686
    
    This was reported in
    
     https://bugzilla.gnome.org/show_bug.cgi?id=701381
    
    Signed-off-by: David Zeuthen <zeuthen@gmail.com>

diff --git a/src/disks/gducreatediskimagedialog.c b/src/disks/gducreatediskimagedialog.c
index 305707a..0708730 100644
--- a/src/disks/gducreatediskimagedialog.c
+++ b/src/disks/gducreatediskimagedialog.c
@@ -613,7 +613,7 @@ copy_span (int              fd,
                                   error))
     {
       g_prefix_error (error,
-                      "Error writing %" G_GUINT64_FORMAT " bytes to offset %" G_GUINT64_FORMAT ": ",
+                      "Error writing %" G_GSIZE_FORMAT " bytes to offset %" G_GUINT64_FORMAT ": ",
                       num_bytes_to_write,
                       offset);
       goto out;
diff --git a/src/disks/gducreatefilesystemwidget.c b/src/disks/gducreatefilesystemwidget.c
index c0b38e3..b8b4f84 100644
--- a/src/disks/gducreatefilesystemwidget.c
+++ b/src/disks/gducreatefilesystemwidget.c
@@ -429,7 +429,7 @@ populate (GduCreateFilesystemWidget *widget)
   if (widget->drive != NULL && udisks_drive_get_removable (widget->drive))
     {
       /* default FAT for flash and disks/media smaller than 20G (assumed to be flash cards) */
-      if (is_flash (widget->drive) || udisks_drive_get_size (widget->drive) < 20L * 1000L*1000L*1000L)
+      if (is_flash (widget->drive) || udisks_drive_get_size (widget->drive) < (guint64)(20ULL * 1000ULL*1000ULL*1000ULL))
         {
           gtk_combo_box_set_active_id (GTK_COMBO_BOX (widget->type_combobox), "vfat");
         }
diff --git a/src/disks/gduformatdiskdialog.c b/src/disks/gduformatdiskdialog.c
index 06dadee..32fce3d 100644
--- a/src/disks/gduformatdiskdialog.c
+++ b/src/disks/gduformatdiskdialog.c
@@ -293,7 +293,7 @@ populate_partitioning_combobox (FormatDiskData *data)
   /* Default to MBR for removable drives < 2TB... GPT otherwise */
   if (data->drive != NULL &&
       udisks_drive_get_removable (data->drive) &&
-      udisks_drive_get_size (data->drive) < 2 * 1000L*1000L*1000L*1000L)
+      udisks_drive_get_size (data->drive) < (guint64)(2ULL * 1000ULL*1000ULL*1000ULL*1000ULL))
     {
       gtk_combo_box_set_active_id (GTK_COMBO_BOX (data->type_combobox), "dos");
     }
diff --git a/src/disks/gdupartitiondialog.c b/src/disks/gdupartitiondialog.c
index 39ca2b3..be00bc0 100644
--- a/src/disks/gdupartitiondialog.c
+++ b/src/disks/gdupartitiondialog.c
@@ -91,11 +91,11 @@ edit_partition_get (EditPartitionData   *data,
       if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (data->bootable_checkbutton)))
         flags |= (1UL<<2);
       if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (data->readonly_checkbutton)))
-        flags |= (1UL<<60);
+        flags |= (1ULL<<60);
       if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (data->hidden_checkbutton)))
-        flags |= (1UL<<62);
+        flags |= (1ULL<<62);
       if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (data->do_not_automount_checkbutton)))
-        flags |= (1UL<<63);
+        flags |= (1ULL<<63);
     }
   else if (g_strcmp0 (data->partition_table_type, "dos") == 0)
     {
@@ -252,9 +252,9 @@ edit_partition_populate (EditPartitionData *data)
       flags = udisks_partition_get_flags (data->partition);
       gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->system_checkbutton),           (flags & (1UL<< 0)) != 0);
       gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->bootable_checkbutton),         (flags & (1UL<< 2)) != 0);
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->readonly_checkbutton),         (flags & (1UL<<60)) != 0);
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->hidden_checkbutton),           (flags & (1UL<<62)) != 0);
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->do_not_automount_checkbutton), (flags & (1UL<<63)) != 0);
+      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->readonly_checkbutton),         (flags & (1ULL<<60)) != 0);
+      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->hidden_checkbutton),           (flags & (1ULL<<62)) != 0);
+      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->do_not_automount_checkbutton), (flags & (1ULL<<63)) != 0);
     }
   else if (g_strcmp0 (data->partition_table_type, "dos") == 0)
     {
diff --git a/src/disks/gdurestorediskimagedialog.c b/src/disks/gdurestorediskimagedialog.c
index 5f64273..46479ae 100644
--- a/src/disks/gdurestorediskimagedialog.c
+++ b/src/disks/gdurestorediskimagedialog.c
@@ -769,7 +769,7 @@ copy_thread_func (gpointer user_data)
                                     &error))
         {
           g_prefix_error (&error,
-                          "Error reading %" G_GUINT64_FORMAT " bytes from offset %" G_GUINT64_FORMAT ": ",
+                          "Error reading %" G_GSIZE_FORMAT " bytes from offset %" G_GUINT64_FORMAT ": ",
                           num_bytes_to_read,
                           num_bytes_completed);
           goto out;
@@ -777,7 +777,7 @@ copy_thread_func (gpointer user_data)
       if (num_bytes_read != num_bytes_to_read)
         {
           g_set_error (&error, G_IO_ERROR, G_IO_ERROR_FAILED,
-                       "Requested %" G_GUINT64_FORMAT " bytes from offset %" G_GUINT64_FORMAT " but only read %" G_GUINT64_FORMAT " bytes",
+                       "Requested %" G_GSIZE_FORMAT " bytes from offset %" G_GUINT64_FORMAT " but only read %" G_GSIZE_FORMAT " bytes",
                        num_bytes_read,
                        num_bytes_completed,
                        num_bytes_to_read);
@@ -792,7 +792,7 @@ copy_thread_func (gpointer user_data)
             goto copy_write_again;
 
           g_set_error (&error, G_IO_ERROR, G_IO_ERROR_FAILED,
-                       "Error writing %" G_GUINT64_FORMAT " bytes to offset %" G_GUINT64_FORMAT ": %m",
+                       "Error writing %" G_GSIZE_FORMAT " bytes to offset %" G_GUINT64_FORMAT ": %m",
                        num_bytes_read,
                        num_bytes_completed);
           goto out;
